### DISCUSSION TITLE
can copilot for enterprise be trained on models specific to team repos?

### DISCUSSION THREAD

**Discussion by francisbabu** [community/community#114462] [Enterprise]
We are planning to move to GitHub enterprise and use Copilot. We have various teams working on different technologies including a few niche techs. Our concern is that if we move all our repos, the Copilot could hallucinate and generate irrelevant codes. So is there a way to have a separate model for each team or technology to train and get the benefits quickly?

**Comment by queenofcorgis** [comment]
Hey @francisbabu, 

Thanks for your patience. You do have the ability to create Knowledge Bases that are limited to specific repos as an Enterprise Admin. [This doc article](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/copilot-chat-in-github/managing-copilot-knowledge-bases) has more details to get you started ✈️ .

**Reply by francisbabu** [reply]
Hi @queenofcorgis, Thank you for your response. I have gone through the document. I think the KB are good to answer common questions about the project like best practices, deployment procedure, etc. But, I am looking for a feature in Copilot where we can train Copilot based on projects, so that we can get the coding responses relevant to the project. 
Ex: Our org has projects that has Next.Js and Shopify. So Copilot is good at giving Next.js code but not at Shopify code.

I don't know how Copilot learns overtime and the ways we can speed up its learning so that we can increase the accuracy of Shopify/SFCC or any other platform-specific codes.
Thank you.
